### PR TITLE
✨ Add a few more ENV settings

### PR DESCRIPTION
**Pull request by dudeofawesome**
- `self_hostname`: This one is particularly useful since you'll need to set it to communicate with the processes running in this container.
- `upnp_disabled`: Allows disabling of UPNP
- `log_level`: Allows setting the log level. This is extremely useful for debugging purposes.

### CODE REVIEW

File: Dockerfile [hunk 1]
```diff
@@ -11,6 +11,7 @@ ENV farmer_address="null"
 ENV farmer_port="null"
 ENV testnet="false"
 ENV full_node_port="null"
+ENV self_hostname="127.0.0.1"
```
Comment (webees) [inline Dockerfile:14]: Need to add "log_level"

**Review by webees — COMMENTED** [review]
Please update as soon as possible!

File: Dockerfile (hunk 1 above)
Comment (cmmarslender) [inline Dockerfile:14]: In what situations is 127.0.0.1 not working for self_hostname?

**Comment by cmmarslender** [comment]
Closing this PR - log_level was introduced in another PR, and there are outstanding conflicts/questions on this one

**Comment by Artanicus** [comment]
Both remaining items are definitely worth implementing.

- `upnp_disabled`: In most Docker setups UPnP could either be completely useless or an active attack surface. If a config option is not desired, it should be disabled by default with the Docker image.
- `self_hostname`: Since this is used to set the address RPC services listen on, any metrics gathering using RPC interfaces will in-fact be coming from a sister container and not from within the container itself. Thus allowing convenient overriding of the listen address allows for more complex monitoring setups, such as using the `chia_exporter` project to export Prometheus metrics from running daemons via the RPC interface. It would be much preferable to allow setting the listen addresses individually per service but since the codebase is right now very heavily tied to `self_hostname` that's probably way more effort and should be thought of as a v2 improvement.

TL;DR; both serve great value in allowing more security, flexibility and enabling the community to extend features based on a solid official building block.

File: Dockerfile (hunk 1 above)
Comment (Artanicus) [reply Dockerfile:14]: Since this is used to set the address RPC services listen on, any metrics gathering using RPC interfaces will in-fact be coming from a sister container and not from within the container itself.
